### DISCUSSION TITLE
API to read job summary

### DISCUSSION THREAD

**Discussion by mpelesh** [community/community#27649] [Apps, API and Webhooks]
Job summary is a great new feature.
Is there a REST API to read job summary?

**Comment by Gisgar3** [comment]
As far as I know, there is no current way to fetch job summaries from the REST API. Try looking [here](https://docs.github.com/en/rest/actions/workflow-jobs#get-a-job-for-a-workflow-run) for more information on getting data regarding workflow jobs! I hope this helps :smiley:

**Reply by ecout** [reply]
Since the Workflow run in my use case is an API triggered workflow_dispatch, I need to verify that the inputs of that event are correct via API call in order to query the dispatch.
IDEALLY, the workflow_dispatch event POST request should return a body containing data with the created workflow run, but it simply returns a 204, queues the request and only later on creates a workflow run for it.
We have to go through lot of request then to match the proper workflow run to the workflow_dispatch event. The API that you pointed out, lacks that data. The only thing I can think of now is annotations.

**Comment by ecout** [comment]
+1:

Please add this as it's another gap for integration(as discussed in the blog post below).

Documentation:
https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary

If there's a write at the action level, there should be a read at the API level.
Thanks!

**Comment by wonglkd** [comment]
Can't find this in the API, whether REST or GraphQL, but the raw Markdown can be accessed at `/<owner>/<repo>/actions/runs/<run>/jobs/<job>/summary_raw`, and someone made a GitHub action to generate this URL: https://github.com/pl-strflt/job-summary-url-action/tree/main

You do need to be logged in to access the summary, though.

**Reply by KyleKing** [reply]
Unfortunately, the `/summary_raw` wasn't working for me. It looks like GitHub has removed raw markdown and the linked GitHub Action returns a link to the Server Side Rendered webpage: `/<owner>/<repo>/actions/runs/<run>/attempts/<attempt>#summary-<summary>`

Example: https://github.com/libp2p/test-plans/actions/runs/5325838866/attempts/1#summary-14413504433 from: https://github.com/libp2p/test-plans/blob/3c7d726d4afaf3e41f1fc1bf27b3bd15baaa761c/.github/workflows/update-badge.yml#L43-L54 doesn't have content at: https://github.com/libp2p/test-plans/actions/runs/5325838866/attempts/1/summary_raw

**Reply by ilyamt-tandemg** [reply]
How about [Render a Markdown document in raw mode](https://docs.github.com/en/rest/markdown/markdown?apiVersion=2022-11-28#render-a-markdown-document-in-raw-mode)?

**Reply by KyleKing** [reply]
@ilyamt-tandemg can you clarify how that would help?

**Reply by koppor** [reply]
Seems to  be unavailable as of today: https://github.com/ipdxco/job-summary-url-action/issues/17

**Comment by david-morris** [comment]
So I'm looking for an API to read the job summary, but that's not what I want to do.

All I want to do is to show the current commit's job summary or a placeholder on the PR page itself.

Shouldn't that be default?  Like, you would see a "show summary" foldout on each of the completed checks?
![image](https://github.com/community/community/assets/5530960/4c784602-867e-4035-827b-b59ce945b1b3)

**Comment by austenstone** [comment]
I made an action [austenstone/job-summary](https://github.com/austenstone/job-summary)

```yml
      - uses: austenstone/job-summary@main
        id: job-summary
        with:
          create-pdf: false
      - run: echo "${{ steps.job-summary.outputs.job-summary }}"
```

**Reply by theamk** [reply]
Info for others: this action works by looking at the parent dir of GITHUB_STEP_SUMMARY file and reading all the files there.
This means: (1) it only works within the current job, no fetching summaries from other jobs and (2) it only works while job is running.
It does, however upload the results as artifact so it can be downloaded later using artifact API.

It's a smart workaround, as long as you don't forget to add it to the end of each job definition you have.

**Reply by felickz** [reply]
💯 exactly what I was looking for so I could upload the summary to an artifact
